decidim-proposals/app/helpers/decidim/proposals/proposal_votes_helper.rb

```diff
@@ -50,6 +50,7 @@ def threshold_per_proposal_enabled?
       #
       # Returns an Integer with the maximum amount of votes, nil otherwise.
       def threshold_per_proposal
+        return 0 if component_settings.threshold_per_proposal.blank?
```

Comment:
So if the threshold if explicitly `0` we'll return `nil`, and if it's explicitly `nil` we'll return `0`? This is very weird, isn't it? Could we add a spec to show the bug? My first intuition would be that a `zero` default in the schema would do the job but it looks like it doesn't...

Reply:
The schema also has a default `zero` for new proposal components created. 
The main problem of this, is for the existing proposal features or components, that the first name of this field was `maximum_votes_per_proposal` and we renamed to `threshold_per_proposal``
For some reason in the proposal features the field `threshold_per_proposal` not exists...

If you have some ideas on how to solve it, I will apreciate it

Reply:
Are we sure the name migration has been properly applied? If the field doesn't exist, maybe it means that the key in the `jsonb` hash has not been migrated. The DB should be double checked directly. At first glance, it might be the data (and thus maybe the migration too), not the code, that needs a fix.

Reply:
As I know, for now, it happens in decidim barcelona. The migration exists in the repo https://github.com/AjuntamentdeBarcelona/decidim-barcelona/blob/master/db/migrate/20180319095060_rename_maximum_votes_per_proposal_to_threshold_per_proposal.decidim.rb 

But I can't not be sure if this migration has been properly applied, because I don't have access to DDBB.

Maybe someone of @decidim/lot-core can help us to take a look into DDBB.

Reply:
@isaacmg410 the migration has been applied, but `threshold_per_proposal` is `nil`.

Wouldn't it be easier to change line 54 and do something like:

```ruby
return nil unless component_settings.threshold_per_proposal.to_i.positive?
```

Reply:
I guess the previous code worked when `maximum_votes_per_proposal` was nil or not defined.

Reply:
@oriolgual What's the situation exactly? If `threshold_for_proposal` in `nil`, I think it means there's no `threshold_for_proposal` key in the settings jsonb hash, and I think that means the migration is incorrect even if it has been applied.

Reply:
To further elaborate, the settings hash should always have all the keys defined in the schema so that attribute defaults can be properly applied, and that's the way it has always worked, I think.

Reply:
For the previous code I only change the  methods name. And create the migration to change the json key name. The value should be the same than the previous value.

Reply:
The key is present but the value is `nil` @deivid-rodriguez

Reply:
Can we know in some way, if the value before the migration was `nil` too? @oriolgual

Reply:
Actually no... The settings manifest includes a presence validation on all attributes, and the field is set to the setting's default by default. The setting should never be `nil` so it seems like a problem with the data, which most likely goes back to before the setting rename since the migration seems correct. I would manually fix the data to switch all settings from `null` to `0`.

Reply:
Done!

Reply:
Did you update the DB? Then I think we don't need this PR...

Reply:
I'd rather include a data migration or rake task for installations to fix their data (and monitor this closely to try to find _if_ this creeps in again, and why), rather than including complexity uncovered by tests...

Reply:
@deivid-rodriguez I'll add a PR with a migration. About the changes, I'd keep it, otherwise `maximum_votes` would crash since `threshold_per_proposal` could return `nil`

Reply:
@oriolgual I don't think so, both methods changed look directly in the component settings, and I don't think the setting can ever be `nil`... :thinking: